wc-vt-giftcards.php

```diff
@@ -458,6 +501,10 @@ public function init_form_fields() {
           $cards = WC()->session->get('gift_cards') ?: array();
           if (isset($cards[$code])) {
             wc_add_notice('That gift card has already been applied to the cart.');
+          }
+          // verify PIN entered
+          elseif (empty($_POST['gift_card_pin'])) {
+            wc_add_notice('Please supply a valid PIN.');
```

Comment:
Does it work to concat the `$_POST['gift_card_pin']` to the `$code` right after the `$is_gift_card` check and let the rest of the code continue as normal? If you did that I think you could remove several other changes, including the ones in `$api_call`